tests/integration/helpers.py

```diff
@@ -266,3 +267,169 @@ def get_job_invoked_unit(job: jenkins.jenkinsapi.job.Job, units: typing.List[Uni
         if unit.name.replace("/", "-") == invoked_agent:
             return unit
     return None
+
+
+def gen_test_pipeline_with_custom_script_xml(script: str):
+    """Generate a job xml with custom pipeline script.
+
+    Args:
+        script: Custom pipeline script.
+
+    Returns:
+        The job XML.
+    """
+    return textwrap.dedent(
+        f"""
+        <flow-definition plugin="workflow-job@1385.vb_58b_86ea_fff1">
+            <actions/>
+            <description></description>
+            <keepDependencies>false</keepDependencies>
+            <properties/>
+            <definition
+                class="org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition"
+                plugin="workflow-cps@3837.v305192405b_c0">
+                <script>{script}</script>
+                <sandbox>true</sandbox>
+            </definition>
+            <triggers/>
+            <disabled>false</disabled>
+        </flow-definition>
+        """
+    )
+
+
+def kubernetes_test_pipeline_script() -> str:
+    """Generate a test pipeline script using the kubernetes plugin.
+
+    Return:
+        The pipeline script
+    """
+    return textwrap.dedent(
+        """
+        podTemplate(yaml: '''
+            apiVersion: v1
+            kind: Pod
+            metadata:
+            labels:
+                some-label: some-label-value
+            spec:
+            containers:
+            - name: httpd
+              image: httpd
+              command:
+              - sleep
+              args:
+              - 99d
+              tty: true
+        ''') {
+        node(POD_LABEL) {
+            stage('Integration Test') {
+            sh '''#!/bin/bash
+                hostname
+            '''
+            }
+        }
+        }"""
+    )
+
+
+def create_secret_file_credentials(
+    unit_web_client: UnitWebClient, kube_config: str
+) -> typing.Optional[str]:
+    """Return the necessary components to create a credentials using the jenkins API.
```

Comment:
```suggestion
    """Return the necessary components to create credentials using the jenkins API.
```

Reply:
I'll update the doc string entirely since this method now works differently